Player Controls

3Ustin opened the pull request:
This PR solves issue #7 .

This is a first implementation of player controls to give functionality to the user. Simple controls are needed sooner than later more sophisticated controls. Volume, seeking, and lastly certain web browsers prevent loading so using activate element will be needed.

3Ustin committed:
clicking functionality with button

the button will be used to display information about current player object when clicked.

3Ustin committed:
State Hook Problem Review

Implementing the useState hook to store and update the player object. Method not permanent.

3Ustin committed:
Review Commit

Current_track.artists[0].name is added to line 50 in WebPlayBack.tsx two issues arise.
1: Multiple instances of Player Object.
2: Deconstructing TypeScript Error.

3Ustin committed:
finished functionality to issue #6

An interface was created so that individualized states could be consolidated. isPlayerGood was removed with playerinfo for less state usage. Button was there for testing and has now been removed for build.

3Ustin committed:
Player Controls

This is an implemented controller system for beginning player controls. The team created a Backward, Toggle, and forward button for easy to start with player functionality.

MtuMartian commented on src/components/WebPlayBack.tsx:
```diff
@@ -1,3 +1,4 @@
+import { Console } from 'console';
```
Are we using this import? We should be able to just call `console.log` from anywhere

3Ustin replied:
No the Team was not. Thank you.

3Ustin replied:
🦍

MtuMartian commented on src/components/WebPlayBack.tsx:
```diff
@@ -46,16 +47,17 @@ function WebPlayback(props: any) {
             });
-
+            //listener to know if Spotify License has been expired
+            //NOT IMPLEMENTED YET
```
Good call! I'll work on creating issues for getting our auth flow set up

MtuMartian commented on src/components/WebPlayBack.tsx:
```diff
@@ -64,20 +66,56 @@ function WebPlayback(props: any) {
         setPlayerInfo(playerInfo);
     }
+    //Method for playing and pausing music based on player state.
+    const toggle = (event: React.MouseEvent<HTMLElement>) => {
```
I like the use of typing here!

3Ustin replied:
Thank you!

MtuMartian commented on src/components/WebPlayBack.tsx:
```diff
@@ -64,20 +66,56 @@ function WebPlayback(props: any) {
+                return;
+            }
+            //@ts-ignore
+            statePlayer.togglePlay().then(() => {
```
I don't think we need to get the current state to call togglePlay. We can probably simplify this by just calling `statePlayer.togglePlay` immediately without getting the current state first. This should probably look more like the `forward` and `backward` functions below

3Ustin replied:
This has been fixed thanks for the hints.

MtuMartian commented on src/components/WebPlayBack.tsx:
```diff
@@ -64,20 +66,56 @@ function WebPlayback(props: any) {
+    const forward = (event: React.MouseEvent<HTMLElement>) => {
+        //@ts-ignore
+        statePlayer.nextTrack().then(() => {
+            console.log('Skipped to next track!');
```
I like the use of the callback to log information! Once we start deploying this app we'll have to rethink how we're logging but this is good for now!

One thing to consider: we could also add a .catch in case the promise rejects and log the error it rejects with

3Ustin replied:
I didn't fix this. I like the way you are thinking I hadn't thought of that Thanks!

MtuMartian reviewed (commented):
Overall looks like a good start!

I left a few comments on things that can be improved. Let's come back to this after #14 is merged

3Ustin committed:
Fixed PR issues

Removed access token for saftey, and capsulated buttons into separate PlayerUI component.

MtuMartian commented on src/App.tsx:
```diff
@@ -2,6 +2,7 @@ import React from 'react';
 import logo from './logo.svg';
 import './App.css';
 import WebPlayback from './components/WebPlayBack';
+import PlayerUI from './components/PlayerUI'
```
This import can be removed

MtuMartian reviewed (approved):
Looks great! Thanks for making those changes :)

3Ustin committed:
Merge branch 'main' into player-controls
